cmd/concourse/validator/validator.go

```diff
@@ -0,0 +1,397 @@
+package validator
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/concourse/baggageclaim/baggageclaimcmd"
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/atc/api/accessor"
+	"github.com/concourse/concourse/atc/atccmd"
+	"github.com/concourse/concourse/atc/wrappa"
+	"github.com/concourse/concourse/skymarshal/skycmd"
+	"github.com/concourse/concourse/tracing"
+	"github.com/concourse/concourse/worker/workercmd"
+	"github.com/concourse/flag"
+	ut "github.com/go-playground/universal-translator"
+	"github.com/go-playground/validator/v10"
+	en_translations "github.com/go-playground/validator/v10/translations/en"
+	"gopkg.in/yaml.v2"
+)
+
+// NewValidator constructs a new validator that contains all the default and
+// custom validations that are necessary to run against the concourse config
+// fields. These include validations on the atc, tsa, worker and baggageclaim
+// fields.
+func NewValidator(trans ut.Translator) *validator.Validate {
```

Comment:
I added `go-playground/validator` to concourse because I wanted to try to consolidate as much validation as I could into one place. Previously, we just had the validation of configuration fields be spread out throughout the code. With this library, you would just add the `validate` tag to the struct field that you want to run a validation against, and you can use the built in validations that the library contains or you can add custom ones. 

I also used it to add custom error messages for the failure of the custom validations. I think that will be nice for showing a prettier error message to the user. I'd love any feedback on this though, not sure if this is better than before or not.